components/elm/src/data_types/GridcellDataType.F90

```diff
@@ -106,31 +106,57 @@ module GridcellDataType
   ! Define the data structure that holds carbon state information at the gridcell level.
   !-----------------------------------------------------------------------
   type, public :: gridcell_carbon_state
-    real(r8), pointer :: seedc          (:) => null()   ! (gC/m2) pool for seeding new PFTs via dynamic landcover
-    real(r8), pointer :: begcb          (:) => null()   ! carbon mass, beginning of time step (gC/m**2)
-    real(r8), pointer :: endcb          (:) => null()   ! carbon mass, end of time step (gC/m**2)
-    real(r8), pointer :: errcb          (:) => null()   ! carbon balance error for the timestep (gC/m**2)
-
+    real(r8), pointer :: seedc                 (:) => null() ! (gC/m2) pool for seeding new PFTs via dynamic landcover
+    real(r8), pointer :: tcs_month_beg         (:) => null()  ! grc total carbon storage at the beginning of a month
+    real(r8), pointer :: tcs_month_end         (:) => null()  ! grc total carbon storage at the end of a month
+    real(r8), pointer :: begcb                 (:) => null() ! carbon mass, beginning of time step (gC/m**2)
```

Comment:
Searching the rest of the code, `grc_cs%begcb` and `grc_cs%beg_totc` appear to always be the equivalent.  
 `grc_cs%begcb`  is only modified in  [BeginGridCBalanceBeforeDynSubgridDriver](https://github.com/E3SM-Project/E3SM/blob/b39f885b293bb3f44aca85c942c7404d0934e8fd/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L996-L1000) with `c2g`.
Likewise, `grc_cs%beg_totc` is only modified in [BeginGridCBalance](https://github.com/E3SM-Project/E3SM/blob/b39f885b293bb3f44aca85c942c7404d0934e8fd/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L808-L809)  with an equivalent call to `c2g` since `col_cs%totcolc` is not modified in between the two calls.   

So one of the fields looks like it can be deleted.  I think `grc_cs%beg_totc` is a better name and more widely used so I would advocate for deleting `begcb`.

I think the same could be done with `grc_cs%endcb` since `c2g` overwrites the `garr`, and `EndGridCBalanceAfterDynSubgridDriver` would need to be changed to use `grc_cs%end_totc`